### PR TITLE
Some quality improvements

### PR DESCRIPTION
**Pull request by cmouse**
- Reduce usage of atoi et al
- Remove boost::lexical cast
- Enforce some constraints more strictly

### CODE REVIEW

**Comment by cmouse** [comment]
Ready for review

**Comment by pieterlexis** [comment]
Looks good to me, mostly automated replacements.

**Comment by mind04** [comment]
I don't like the enforcing of uint16_t values at the cost of a perfectly good backed connection. Out of range is detected before we put the records in a packet.
Are you sure all std::out_of_range exceptions are caught as you are replacing all non throwing atoi() calls by throwing pdns_stou() ones.

**Comment by cmouse** [comment]
It's very hard to be sure, but during testing I had some cases where this was thrown for bad reason and those were caught.

**Comment by Habbie** [comment]
@cmouse can you please try replying to mind04 again? Your answer made zero sense to me.

**Comment by cmouse** [comment]
It is not easy to be 100% certain that the exceptions are caught. However, during making this patch I had several occasions where exception was raised, and those did get caught.

The uint16_t enforcement can be removed if so wanted.

**Comment by cmouse** [comment]
Also, in pdns/distributor.hh:209 we catch *all* exceptions, except PDNSException, and proceed on.

**Comment by Habbie** [comment]
If this PR changes 'integer out of range' from 'domain does not work' to 'DB gets disconnected', then yes, please revert that.

**Comment by cmouse** [comment]
I changed them to just use pdns_stou, so no range check for uint16_t.

**Comment by Habbie** [comment]
Thanks! LGTM

File: pdns/backends/gsql/gsqlbackend.cc [hunk 1]
```diff
@@ -1591,9 +1592,12 @@ void GSQLBackend::extractRecord(const SSqlStatement::row_t& row, DNSResourceReco
 
   r.qtype=row[3];
 
-  if (r.qtype==QType::MX || r.qtype==QType::SRV)
+  if (r.qtype==QType::MX || r.qtype==QType::SRV) {
+    auto prio = std::stoul(row[2]);
+    if (prio > UINT16_MAX)
+      throw std::range_error((boost::format("priority value %lu for %s is too large") % prio % row[0]).str());
```
Comment (mind04) [inline pdns/backends/gsql/gsqlbackend.cc]: missed a spot

Comment (cmouse) [inline pdns/backends/gsql/gsqlbackend.cc]: thanks *duh*

**Pull request merged**